File: LISTA-03(JAVA)/Q1.java
class Retangulo {
    private double b = 0;
    private double h = 0;
    public Retangulo(double b, double h) {
        this.setBase(b);
        this.setAltura(h);
    }

    public void setBase(double v) {
        if (v >= 0) this.b = v;
        else throw new IllegalArgumentException("valor inválido");
    }
    public void setAltura(double v) {
        if (v >= 0) this.h = v;
        else throw new IllegalArgumentException("valor inválido");
    }
    public double getBase() {
        return this.b;
    }
    public double getAltura() {
        return this.h;
    }
    
    public double calcArea() {
        return this.b * this.h;
    }
    public double calcDiagonal() {
        return (this.b * this.b) + (this.h * this.h);
    }
    public String toString() {
        return "Base =" + this.b + "- Altura = " +  this.h;
    }
}


public class Q1 {
    public static void main(String[] args) {
        Retangulo x = new Retangulo(10, 20);
        System.out.println(x);
        System.out.println(x.calcArea());
    }
    
}

File: LISTA-03(PYTHON)/Q1.py
class Retangulo :
    def __init__(self, b, h) :
        self.SetBase(b)
        self.SetAltura(h)
    def SetBase(self, v) :
        if v >= 0 :
            self.__b = v
        else :
            raise ValueError()
    def SetAltura(self, v) :
        if v >= 0 :
            self.__h = v
        else :
            raise ValueError()
    def GetBase(self) :
        return self.__b
    def GetAltura(self) :
        return self.__h
    def CalcArea(self) :
        return self.__b * self.__h
    def CalcDiagonal(self) :
        return self.__b**2 + self.__h**2
    def __str__(self) :
        return f"Base = {self.__b} - Altura = {self.__h}"
    
class UI :
    @staticmethod
    def main():
        x = Retangulo(10, 10)
        print(f"Dados do meu retângulo: {x.__str__()}")
        print(f"A área é: {x.CalcArea()}. A diagonal é: {x.CalcDiagonal()}")

UI.main()